standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1513,6 +1516,11 @@ public void commitTxn(CommitTxnRequest rqst)
         } else if (txnType.get() == TxnType.COMPACTION) {
           acquireTxnLock(stmt, false);
           commitId = getHighWaterMark(stmt);
+        } else if (txnType.get() != TxnType.READ_ONLY && isInsert(stmt, insertFilter)) {
```

Comment:
We should ignore repl as well. Please extract "txnType.get() != TxnType.READ_ONLY && !rqst.isSetReplPolicy()" nand handle inserts and updates there.